samples/dotnet/kernel-syntax-examples/Example41_Weaviate.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Globalization;
+using System.Threading.Tasks;
+using Microsoft.SemanticKernel;
+using Microsoft.SemanticKernel.Connectors.Memory.Weaviate;
+using Microsoft.SemanticKernel.Memory;
+using RepoUtils;
+
+// ReSharper disable once InconsistentNaming
+public static class Example41_Weaviate
+{
+    private const string MemoryCollectionName = "weaviate-test";
+
+    public static async Task RunAsync()
+    {
+        string scheme = Env.Var("WEAVIATE_SCHEME");
+        string endpoint = Env.Var("WEAVIATE_ENDPOINT");
+        int weaviatePort = int.Parse(Env.Var("WEAVIATE_PORT"), CultureInfo.InvariantCulture);
+        string apiKey = Env.Var("WEAVIATE_APIKEY");
+        WeaviateMemoryStore memoryStore = new WeaviateMemoryStore(scheme, endpoint, weaviatePort, apiKey, logger: ConsoleLogger.Log);
+        IKernel kernel = Kernel.Builder
+            .WithLogger(ConsoleLogger.Log)
+            .Configure(c =>
```

Comment:
I think instead of `Configure` should do something like this:

```
            .WithOpenAITextCompletionService("text-davinci-003", Env.Var("OPENAI_API_KEY"))
            .WithOpenAITextEmbeddingGenerationService("text-embedding-ada-002", Env.Var("OPENAI_API_KEY"))
```

Reply:
I did get it running, but hitting this error in the example:

``
== Printing Collections in DB ==
SKweaviatetest
== Adding Memories ==
== Printing Collections in DB ==
SKweaviatetest
== Retrieving Memories Through the Kernel ==
Unhandled exception. System.Collections.Generic.KeyNotFoundException: The given key 'sk_timestamp' was not present in the dictionary.
   at System.Collections.Generic.Dictionary`2.get_Item(TKey key)
   at Microsoft.SemanticKernel.Connectors.Memory.Weaviate.WeaviateMemoryStore.GetAsync(String collectionName, String key, Boolean withEmbedding, CancellationToken cancel) in D:\repos\semantic-kernel\dotnet\src\Connectors\Connectors.Memory.Weaviate\WeaviateMemoryStore.cs:line 275
   at Microsoft.SemanticKernel.Memory.SemanticTextMemory.GetAsync(String collection, String key, Boolean withEmbedding, CancellationToken cancellationToken) in D:\repos\semantic-kernel\dotnet\src\SemanticKernel\Memory\SemanticTextMemory.cs:line 79   
   at Example41_Weaviate.RunAsync() in D:\repos\semantic-kernel\samples\dotnet\kernel-syntax-examples\Example41_Weaviate.cs:line 57
   at Program.Main() in D:\repos\semantic-kernel\samples\dotnet\kernel-syntax-examples\Program.cs:line 131
   at Program.<Main>()
```

Reply:
Fixed. The timestamp is nullable, yet the code presumed it was not.